core/src/main/java/com/amplifyframework/auth/CognitoCredentialsProvider.kt

```diff
@@ -30,7 +27,7 @@ import kotlin.coroutines.suspendCoroutine
  * Internal implementation of cognito credentials provider.
  * This will be ported to core once it seems feasible to do so.
  */
-internal class CognitoCredentialsProvider : AuthCredentialsProvider {
+open class CognitoCredentialsProvider : AuthCredentialsProvider {
```

Comment:
We dont expect to extend this class, so `open` would not be required.

Reply:
internal object DummyCredentialsProvider : CognitoCredentialsProvider()  needs it hence I made it open. And since I was refactoring I wanted to touch as little files as possible.